### PR TITLE
[RFE#1714] chore: build languagetool as modules

### PR DESCRIPTION
**Pull request by miurahr**
Languagetools language libraries are optional for OmegaT application.
This proposal changes these libraries as module.

In future, we will distribute these modules as optional plugin packages when  #226  merged.

## Pull request type

- Build and release changes -> [build/release]

## Which ticket is resolved?

- #1714 Make language libraries in LanguageTools as module
- https://sourceforge.net/p/omegat/feature-requests/1714/

## What does this PR change?

- Form subproject `languagetools`
- produce modules that names are `omegat-language-XX.jar` in build/modules 

## Other information

[dev-ML discussion](https://sourceforge.net/p/omegat/mailman/message/37833902/)  suggests
> > There is about 300MB with dependencies. We change core dependnecies such as LanguageTools and Apache Lucene.
>> 
>> LanguageTool 3.5 takes 65MB,  and LanguageTool 5.7 takes 129MB.

> 
> Would it be possible to download LanguageTools separately (from within OmegaT) ?
> 
> LanguageTool in not a core OmegaT function. A lot of languages do not have use for it.
> 
> For first users, we could have a dialog that says "if you use such and such language, install LanguageTool to have access to grammatical... checks" and provide a link.
> 
> Lucene is a core function but not all languages have specific tokenizers, so here again, it there is a need to download the whole thing, do something similar for users who could benefit from Lucene.
>

### CODE REVIEW

**Comment by miurahr** [comment]
Review points   file `languagetools/build.gradle`
- Plugin name and description template
- Language name

**Comment by miurahr** [comment]
> Would it be possible to download LanguageTools separately (from within OmegaT) ?

To allow download  LT separately, #226 is required.

**Comment by miurahr** [comment]
TODO: 

- [ ] Bundle language libraries in source distribution. currently only `core` library is there.

File: build.gradle [hunk 1]
```diff
@@ -566,12 +559,12 @@ distributions {
...
-            into('lib/provided') {
-                // collect subproject runtime dependencies
-                from  {subprojects.findAll { it.getSubprojects().isEmpty()}.
-                        collect { it.configurations.runtimeClasspath }
+                // collect project runtime dependencies in all subprojects and sourceSets
+                // Because subproject has dependency for root project, so we don't add
+                // rootProject's dependencies to avoid duplicate.
+                // from configurations.runtimeClasspath
+                from {subprojects.findAll { it.getSubprojects().isEmpty()}
+                        .collect { it.configurations.matching { it.name.endsWith('untimeClasspath') } }
```
Comment (Kazephil) [inline build.gradle:567]: Is `'untimeClasspath'` a typo for `'runtimeClasspath'`?

File: languagetools/build.gradle [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        [ lang: 'Belarusian', code: 'be', className: 'BelarusianPlugin'],
+        [ lang: 'Breton', code: 'br', className: 'BretonPlugin'],
+        [ lang: 'Catalan', code: 'ca', className: 'CatalanPlugin'],
+        [ lang: 'Danish', code: 'da', className: 'DanishPlugin'],
+        [ lang: 'German', code: 'de', className: 'GermanPlugin'],
+        [ lang: 'Greek', code: 'el', className: 'GreekPlugin'],
+        [ lang: 'English', code: 'en', className: 'EnglishPlugin'],
+        [ lang: 'Esperanto', code: 'eo', className: 'EsperantoPlugin'],
+        [ lang: 'Spanish', code: 'es', className: 'SpanishPlugin'],
+        [ lang: 'Percian', code: 'fa', className: 'PercianPlugin'],
```
Comment (Kazephil) [inline languagetools/build.gradle:21]: Typos:
- Percian --> Persian
- PercianPlugin --> PersianPlugin

This also affects other files.

File: languagetools/src/fa/java/org/omegat/languagetool/language/PercianPlugin.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.omegat.languagetool.language;
+
+public class PercianPlugin {
```
Comment (Kazephil) [inline languagetools/src/fa/java/org/omegat/languagetool/language/PercianPlugin.java:28]: PercianPlugin --> PersianPlugin

File: languagetools/build.gradle [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        [ lang: 'French', code: 'fr', className: 'FrenchPlugin'],
+        [ lang: 'Irish', code: 'ga', className: 'IrishPlugin'],
+        [ lang: 'Galcian', code: 'gl', className: 'GalcianPlugin'],
```
Comment (Kazephil) [inline languagetools/build.gradle:24]: Galcian --> Galician
GalcianPlugin --> GalicianPlugin

File: languagetools/src/gl/java/org/omegat/languagetool/language/GalcianPlugin.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.omegat.languagetool.language;
+
+public class GalcianPlugin {
```
Comment (Kazephil) [inline languagetools/src/gl/java/org/omegat/languagetool/language/GalcianPlugin.java:28]: Galcian --> Galician

**Review by Kazephil — COMMENTED** [review]
I spotted a few typos.

File: build.gradle (hunk 1 above)
Comment (miurahr) [reply build.gradle:567]: It is not a typo. It try to catch "enRuntimeClasspath", "ruRuntimeClasspath",... (all languages ) and normal "runtimeClassPath".

Comment (miurahr) [reply build.gradle]: Pls see amended comment

**Review by Kazephil — APPROVED** [review]
Looks good to me now.

**Comment by miurahr** [comment]
I found that a source distribution has unwanted library jar files. It may be caused by a filter of "*RuntimeClasspath" that may include dependencies of Gradle plugins such as SpotBugs dependency.
It breaks build process of source distribution.

**Comment by miurahr** [comment]
> I found that a source distribution has unwanted library jar files. It may be caused by a filter of "*RuntimeClasspath" that may include dependencies of Gradle plugins such as SpotBugs dependency. It breaks build process of source distribution.

This caused from dependency jStyleParser library. I'd like to fix the issue in another PR.

**Pull request merged**